modules/data-entry/src/main/frontend/src/questionnaireEditor/AnswerOptions.jsx

```diff
@@ -84,6 +86,24 @@ let AnswerOptions = (props) => {
     }
   ]
 
+  let getItemStyle = (isDragging, draggableStyle) => ({
+    // change background colour if dragging
+    border: isDragging ? "2px dashed" : "none",
+    borderColor: isDragging ? "lightblue" : "",
+    ...draggableStyle
+  });
+
+  let onDragEnd = (result) => {
+    // dropped outside the list
+    if (!result.destination) {
+      return;
+    }
+    let oldOptions = options.slice();
+    const [removed] = oldOptions.splice(result.source.index, 1);
+    oldOptions.splice(result.destination.index, 0, removed);
+    setOptions(oldOptions);
+  }
+
   // Clear local state when data changes
   useEffect(() => {
     setOptions(Object.values(data).filter(value => value['jcr:primaryType'] == 'lfs:AnswerOption' && !value.notApplicable && !value.noneOfTheAbove).slice());
```

Comment:
This will fix the bug found by @sashaandjic . But the code is the same as on line 44, so we should consider pulling it into a function to avoid duplication.

```suggestion
    setOptions(Object.values(data).filter(value => value['jcr:primaryType'] == 'lfs:AnswerOption' && !value.notApplicable && !value.noneOfTheAbove).sort((option1, option2) => (option1.defaultOrder - option2.defaultOrder)));
```